core/src/main/java/org/elasticsearch/common/io/stream/StreamInput.java

```diff
@@ -543,6 +533,17 @@ public GeoPoint readGeoPoint() throws IOException {
         }
     }
 
+    /**
+     * Serializes a potential null value.
+     */
+    public <T extends StreamableReader<T>> T readOptionalStreamableReader(StreamableReader<T> streamableReader) throws IOException {
```

Comment:
Why not just make `SeqNoStats` implement `Streamable` and just use the existing `readOptionalStreamable` and `writeOptionalStreamable`? That's consistent with most (all?) the existing stats objects.